core/src/main/java/com/linecorp/armeria/common/util/AsyncCloseable.java

```diff
@@ -18,16 +18,28 @@
 import java.util.concurrent.CompletableFuture;
 
 /**
- * An object that may hold resources until it is closed. Unlike {@link AutoCloseable}, the {@link #closeAsync()}
- * method releases the resources asynchronously, returning the {@link CompletableFuture} which is completed
- * after the resources are released.
+ * An object that may hold resources until it is closed. In addition to {@link AutoCloseable#close()},
+ * this interface provides {@link #closeAsync()} which releases the resources asynchronously, returning
+ * a {@link CompletableFuture} which is completed after the resources are released.
  */
-@FunctionalInterface
-public interface AsyncCloseable {
+public interface AsyncCloseable extends AutoCloseable {
     /**
-     * Releases the resources held by this object asynchronously.
+     * Returns the {@link CompletableFuture} which is completed after the resources are released. Note that
+     * you must use {@link #close()} or {@link #closeAsync()} to release the resources actually. This method
+     * merely returns the {@link CompletableFuture}.
+     */
+    CompletableFuture<?> closeFuture();
```

Comment:
Having two similar names with same return type is making me scratch my head a little. It looks like future is a way of closing out of these three

- (Implicitly implied) Blocking close
- Async close
- Future close

I know this matches our general pattern, but shall we consider renaming all `fooFuture` to `whenFoo`? `ctx.log().whenComplete().thenAccept()` reads like English and seems quite nice to me.

Reply:
That's a good idea. Let me rename. `whenClosed()`, then?